presto-main/src/main/java/com/facebook/presto/operator/PagesHashStrategy.java

```diff
@@ -76,6 +76,17 @@
      */
     boolean positionEqualsRow(int leftBlockIndex, int leftPosition, int rightPosition, Page page, int[] rightChannels);
 
+    /**
+     * Compares the hashed columns in this PagesHashStrategy to the hashed columns in the Page. The
+     * values are compared positionally, so {@code rightChannels} must have the same number of entries as
+     * the hashed columns and each entry is expected to be the same type.
+     * The comparision uses IS NOT DISTINCT FROM.
+     */
+    default boolean positionNotDistinctFromRow(int leftBlockIndex, int leftPosition, int rightPosition, Page page, int[] rightChannels)
+    {
+        throw new UnsupportedOperationException();
```

Comment:
Why do we have default implementation here? Other methods (e.g. `rowEqualsRow`) doesn't have it.